oktapam/client/cloud_connection.go

```diff
@@ -0,0 +1,115 @@
+
+type CloudConnectionDetails struct {
+	AccountId  *string `json:"account_id"`
+	ExternalId *string `json:"external_id"`
+	RoleArn    *string `json:"role_arn"`
+}
+
+type CloudConnection struct {
+	DeletedAt              *string                 `json:"deleted_at,omitempty"`
+	ID                     *string                 `json:"id,omitempty"`
+	Name                   *string                 `json:"name"`
+	Provider               *string                 `json:"provider"`
+	CloudConnectionDetails *CloudConnectionDetails `json:"cloud_connection_details"`
+}
+
+func (c CloudConnection) Exists() bool {
+	return utils.IsNonEmpty(c.ID) && utils.IsBlank(c.DeletedAt)
+}
+
+func (c CloudConnection) ToResourceMap() map[string]any {
+	m := make(map[string]any, 2)
+
+	if c.Name != nil {
+		m[attributes.Name] = *c.Name
+	}
+	if c.ID != nil {
+		m[attributes.ID] = *c.ID
+	}
+	if c.DeletedAt != nil {
+		m[attributes.DeletedAt] = *c.DeletedAt
+	}
+	m[attributes.CloudConnectionDetails] = c.CloudConnectionDetails
+
+	return m
+}
+
+func (c OktaPAMClient) GetCloudConnection(ctx context.Context, name string, allowDeleted bool) (*CloudConnection, error) {
+	requestURL := fmt.Sprintf("/v1/teams/%s/cloud_connections/%s", url.PathEscape(c.Team), url.PathEscape(name))
+	logging.Tracef("making GET request to %s", requestURL)
+	resp, err := c.CreateBaseRequest(ctx).SetResult(&CloudConnection{}).Get(requestURL)
+	if err != nil {
+		logging.Errorf("received error while making request to %s", requestURL)
+		return nil, err
+	}
+	statusCode := resp.StatusCode()
+
+	if statusCode == http.StatusOK {
+		cloudConnection := resp.Result().(*CloudConnection)
+		if cloudConnection.Exists() || allowDeleted {
+			return cloudConnection, nil
+		}
+		return nil, nil
+	} else if statusCode == http.StatusNotFound {
+		return nil, nil
+	}
+
+	return nil, createErrorForInvalidCode(resp, http.StatusOK, http.StatusNotFound)
+}
+
+func (c OktaPAMClient) CreateCloudConnection(ctx context.Context, cloudConnection CloudConnection) (*CloudConnection, error) {
+	requestURL := fmt.Sprintf("/v1/teams/%s/cloud_connections", url.PathEscape(c.Team))
+	logging.Tracef("making POST request to %s", requestURL)
+	resultingCloudConnection := &CloudConnection{}
+	resp, err := c.CreateBaseRequest(ctx).SetBody(cloudConnection).SetResult(resultingCloudConnection).Post(requestURL)
+	if err != nil {
+		logging.Errorf("received error while making request to %s", requestURL)
+		return nil, err
+	}
+	if _, err = checkStatusCode(resp, http.StatusCreated); err != nil {
+		return nil, err
+	}
+	return resultingCloudConnection, nil
+}
+
+func (c OktaPAMClient) UpdateCloudConnection(ctx context.Context, resourceGroupID string, cloudConnection CloudConnection) error {
+	requestURL := fmt.Sprintf("/v1/teams/%s/cloud_connections/%s", url.PathEscape(c.Team), url.PathEscape(resourceGroupID))
+	logging.Tracef("making PUT request to %s", requestURL)
+	resp, err := c.CreateBaseRequest(ctx).SetBody(cloudConnection).Put(requestURL)
```

Comment:
do we want to validate things before the PUT request (also for POST and DELETE)?
I am not sure if this is the norm but lets check that!

Reply:
Added